lib/rester/service/resource/params.rb

```diff
@@ -192,6 +192,9 @@ def _validate(key, value, klass, opts)
 
       def _validate_str(key, value, klass, opts)
         fail unless value.is_a?(String) # assert
+        if [Array, Hash].include?(klass) && key.class != klass
+          _error!("#{key} should be a #{klass} but a #{value.class} was found")
+        end
```

Comment:
What's the `key.class != klass` logic about?  That doesn't seem right.  Key will always be a Symbol.  I think you may have meant `value.klass != klass`, but value should always be a string here so we don't want to do that either.

Also, each of the `_validate_*` methods has a similar check at the top. I think the messages should mirror each other.

What about `"expected #{key} to be of type #{klass}"`?  In that case, we should update the messages in `_validate_array` and `_validate_hash` to `"unexpected Array for #{key}"` and `"unexpected Hash for #{key}"`, respectively.